### DISCUSSION TITLE
Retaining build caching on Mac

### DISCUSSION THREAD

**Discussion by max-sixty** [PyO3/pyo3#2335] [Questions] [answered]
We recently implemented the `config.toml` as recommended in https://pyo3.rs/v0.14.5/building_and_distribution.html#macos. 

Unfortunately this disables some part of cargo build caching on Mac as discussed at https://github.com/prql/prql/issues/411. Is anyone familiar with an approach that allows pyo3 to work on Mac without giving up some caching, particularly for other crates in the repo?

Thank you!

**Comment by davidhewitt** [comment]
Is this potentially related to #1708? (i.e. I wonder if it's something like - "rust-analyzer is not using those flags, so it invalidates the cache") If so, the same workaround of setting rust-analyzer to use a separate target dir might help?

**Comment by davidhewitt** [comment, marked as answer]
You might also want to try the `pyo3_build_config` build script documented just above the `config.toml` option?

**Reply by max-sixty** [reply]
This seems to fix it! Thanks @davidhewitt !
